### PR TITLE
releases: add TiDB 5.0 RC release notes

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

add TiDB 5.0 RC release notes

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/4507
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by hunterlxt** [comment]
## TiKV
- [Improvements] 默认启用 compaction guard 功能，以 TiKV region 的边界来分割 RocksDB SST 文件，从而减少总体的 compaction IO 流量。
- [New features] 支持动态更改 RocksDB rate limiter 的自动调整模式。

**Comment by crazycs520** [comment]
# TiDB

## New Features

- 支持 `utf8mb4_unicode_ci` and `utf8_unicode_ci` collation。
  - [project](https://github.com/pingcap/tidb/issues/17596)
  - [document](https://docs.pingcap.com/tidb/dev/character-set-and-collation#new-framework-for-collations)
- 支持错误和日志信息脱敏.
  - [project](https://github.com/pingcap/tidb/issues/18566)
  - [document](https://github.com/pingcap/tidb/blob/master/errno/logredaction.md)
- 支持 Invisible Indexes.
  - [project](https://github.com/pingcap/tidb/issues/9246)
  - [document](https://github.com/pingcap/tidb/pull/15366)

## Experimental Features

- 支持 Async commit. Async Commit 优化了事务提交时间，在所有 prewrite 请求成功执行完后即可返回结果给客户端。
  - [project](https://github.com/tikv/tikv/projects/34)
  - [document](https://github.com/pingcap/docs-cn/pull/5181)
- 支持 clustered Index. 聚集索引可以提高某些查询的性能，例如点查。
  - [project](github.com/pingcap/tidb/projects/45)
  - [document](https://docs.pingcap.com/tidb/dev/clustered-indexes)
- 支持 `LIST` 和 `LIST COLUMNS` 分区表.
  - [project](https://github.com/pingcap/tidb/issues/20678)
  - [document](https://docs.pingcap.com/zh/tidb/dev/partitioned-table#list-%E5%88%86%E5%8C%BA)
- 提高 Index Selection 的准确度和鲁棒性。
  - [#21817](https://github.com/pingcap/tidb/pull/21817)
  - [document](https://github.com/pingcap/docs-cn/pull/5164)

## Improvements

- 收集更多的算子执行信息。
  - [issue](https://github.com/pingcap/tidb/issues/18663)
  - [document](https://docs.pingcap.com/zh/tidb/stable/sql-statement-explain-analyze#explain-analyze)
- 优化批量删除场景下的性能。
  - [issue](https://github.com/pingcap/tidb/issues/18028)

File: releases/release-5.0.0-rc.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+## 兼容性更改
+
+
+
+## 新功能
+
+
+## 改进提升
+
```
Comment (zanmato1984) [inline releases/release-5.0.0-rc.md:19]:
```suggestion
+ TiFlash
    - 增加线程池排队处理 coprocessor 任务以降低执行高并发 coprocessor 时的内存压力
```

File: releases/release-5.0.0-rc.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# TiDB 5.0 RC Release Notes
+
+发版日期：2020 年 12 月 31 日
+
+TiDB 版本：5.0.0-rc
+
+## 兼容性更改
+
+
```
Comment (flowbehappy) [inline releases/release-5.0.0-rc.md:13]:
```suggestion
+ TiFlash
    - 支持限制 DeltaIndex 的内存使用量，避免大数据量下内存使用过多的问题
    - 支持限制后台数据整理任务使用的 IO 写流量，降低对前台任务的影响
    
```

File: releases/release-5.0.0-rc.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        - [project](https://github.com/pingcap/tidb/issues/18566)
+        - [document](https://github.com/pingcap/tidb/blob/master/errno/logredaction.md)
+    - 支持 Invisible Indexes
+        - [project](https://github.com/pingcap/tidb/issues/9246)
+        - [document](https://github.com/pingcap/tidb/pull/15366)
+    - 支持 Async commit. Async Commit 优化了事务提交时间，在所有 prewrite 请求成功执行完后即可返回结果给客户端（实验特性）
+        - [project](https://github.com/tikv/tikv/projects/34)
+        - [document](https://github.com/pingcap/docs-cn/pull/5181)
+    - 支持 clustered Index. 聚集索引可以提高某些查询的性能，例如点查（实验特性）
+        - [project](github.com/pingcap/tidb/projects/45)
```
Comment (TomShawn) [inline releases/release-5.0.0-rc.md:35]:
```suggestion
        - [project](https://github.com/pingcap/tidb/projects/45)
```

**Comment by amyangfei** [comment]
## TiCDC
### Experimental Features
- 默认使用 unified sorter 作为排序引擎 [#1231](https://github.com/pingcap/ticdc/pull/1231)

File: releases/release-5.0.0-rc.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - 支持 clustered Index. 聚集索引可以提高某些查询的性能，例如点查（实验特性）
+        - [project](https://github.com/pingcap/tidb/projects/45)
+        - [document](https://docs.pingcap.com/tidb/dev/clustered-indexes)
+    - 支持 `LIST` 和 `LIST COLUMNS` 分区表（实验特性）
+        - [project](https://github.com/pingcap/tidb/issues/20678)
+        - [document](https://docs.pingcap.com/zh/tidb/dev/partitioned-table#list-%E5%88%86%E5%8C%BA)
+    - 提高 Index Selection 的准确度和鲁棒性（实验特性）
+        - [#21817](https://github.com/pingcap/tidb/pull/21817)
+        - [document](https://github.com/pingcap/docs-cn/pull/5164)
+
```
Comment (HunDunDM) [inline releases/release-5.0.0-rc.md:36]:
```suggestion

+ PD

    - 支持使用 V2 版本 Region 算分公式来提升平衡调度的收敛速度和稳定性。
        - [#3269](https://github.com/tikv/pd/pull/3269)
        - [document](https://docs.pingcap.com/zh/tidb/dev/pd-configuration-file#region-score-formula-version)
    - 支持错误和日志信息脱敏
        - [project](https://github.com/pingcap/tidb/issues/18566)
        - [#2852](https://github.com/tikv/pd/issues/2852)
        - [document](https://docs.pingcap.com/zh/tidb/dev/pd-configuration-file#redact-info-log-%E4%BB%8E-v500-rc-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5)
    - 支持通过 PD 处理 Split Regions 请求 [#2965](https://github.com/tikv/pd/issues/2965)
    - 支持使用 Joint Consensus 来进行 Region 调度。（实验特性）
        - [project](https://github.com/tikv/tikv/issues/7587)
        - [document](https://docs.pingcap.com/zh/tidb/dev/pd-configuration-file#enable-joint-consensus-%E4%BB%8E-v500-rc-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5)

```

File: releases/release-5.0.0-rc.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+## 改进提升
+
++ TiDB
+
+    - 收集更多的算子执行信息。
+        - [issue](https://github.com/pingcap/tidb/issues/18663)
+        - [document](https://docs.pingcap.com/zh/tidb/stable/sql-statement-explain-analyze#explain-analyze)
+    - 优化批量删除场景下的性能。
+        - [issue](https://github.com/pingcap/tidb/issues/18028)
+
```
Comment (HunDunDM) [inline releases/release-5.0.0-rc.md:55]:
```suggestion

+ PD

    - 默认开启 `enable-cross-table-merge` 来减少空 Region 对集群调度的影响。 [#3129](https://github.com/tikv/pd/pull/3129)
    - 默认开启 Placement Rules [#3006](https://github.com/tikv/pd/pull/3006)
    - `pd-ctl` 的 `transfer-region` 操作支持指定 peer role。[#3047](https://github.com/tikv/pd/pull/3047)

```

**Comment by TomShawn** [comment]
This PR is replaced by https://github.com/pingcap/docs-cn/pull/5295.

**Pull request closed without merging**